path_role.go

```diff
@@ -0,0 +1,346 @@
+package jwtauth
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	sockaddr "github.com/hashicorp/go-sockaddr"
+	"github.com/hashicorp/vault/helper/parseutil"
+	"github.com/hashicorp/vault/helper/policyutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+// pathsRole returns the path configurations for the CRUD operations on roles
+func pathsRole(b *jwtAuthBackend) []*framework.Path {
+	return []*framework.Path{
+		&framework.Path{
+			Pattern: "role/?",
+			Callbacks: map[logical.Operation]framework.OperationFunc{
+				logical.ListOperation: b.pathRoleList,
+			},
+			HelpSynopsis:    strings.TrimSpace(roleHelp["role-list"][0]),
+			HelpDescription: strings.TrimSpace(roleHelp["role-list"][1]),
+		},
+		&framework.Path{
+			Pattern: "role/" + framework.GenericNameRegex("name"),
+			Fields: map[string]*framework.FieldSchema{
+				"name": &framework.FieldSchema{
+					Type:        framework.TypeString,
```

Comment:
Can use `framework.TypeLowerCaseString` and eliminate the `ToLower`s elsewhere in this file.

Reply:
Oh yeah!